webapp/lib/controller.dart

```diff
@@ -90,6 +96,78 @@ void initUI() {
   );
 }
 
+Map<String, model.NeedsReplyData> getLatestDataForProjects(List<model.NeedsReplyData> updatedData) {
+  Map<String, model.NeedsReplyData> latestProjectData = {};
+  
+  for (var project in projectList) {
+    var data = updatedData.where((data) => data.project == project).toList();
+    
+    if (data.isEmpty) {
+      latestProjectData[project] = null;
+    } else {
+      latestProjectData[project] = data.last;
+    }
+  }
+  return latestProjectData;
+}
+
+bool isProjectStale(model.NeedsReplyData projectData) {
+  var now = new DateTime.now();
+  int lastUpdateTimeDiff =  now.difference(projectData.datetime).inHours;
+
+  if (lastUpdateTimeDiff >= 1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
+  projectTimers[projectData.project].cancel();
+
+  if (stale) {
+    projectTimers[projectData.project] = null;
+  } else {
+    var timeToExecute =  projectData.datetime.add(Duration(hours: 1));
+    var now = new DateTime.now();
+    var duration = timeToExecute.difference(now);
+    var timer = new Timer(duration, () => {
+        if (projectData.project == view.contentView.projectSelectorView.selectedProject) {
+          view.contentView.stale = true
+        }
+      }
+    );
+    projectTimers[projectData.project] = timer;
+  }
+}
+
+void checkNeedsReplyMetricsStale(List<model.NeedsReplyData> updatedData) {
+  updatedData.sort((d1, d2) => d1.datetime.compareTo(d2.datetime));
+
+  var selectedProjectName = view.contentView.projectSelectorView.selectedProject;
+  Map<String, model.NeedsReplyData> latestDataPerProject = getLatestDataForProjects(updatedData);
+  
+  for (var project in latestDataPerProject.keys) {
+    var projectData = latestDataPerProject[project];
+
+    if (projectData != null) {
+
+      if (isProjectStale(projectData)) {
+        setupProjectTimer(projectData, true);
+      } else {
+        setupProjectTimer(projectData);
+      }
+    }
+  }
+
+  var selectedProjectTimer = projectTimers[selectedProjectName];
+  if (selectedProjectTimer != null) {
```

Comment:
This misbehaves when we receive new data on an unselected project and the selected project is stale but has had a timer expire, so the timer is not null, just inactive. This clause would then turn the selected project from stale to not stale when it shouldn't.

We should use the same logic as below on line 200:
```suggestion
  if (selectedProjectTimer != null && selectedProjectTimer.isActive) {
```